File: samples/kmp/shared/src/commonMain/kotlin/jp/kamusoft/ksdialogs/samples/kmp/TextInputDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.kmp

import jp.kamusoft.ksdialogs.kmp.DialogViewModel

/**
 * Text Input Dialog の ViewModel。
 *
 * 結果型は [String] で、完了操作は入力された文字列を報告する。
 * 共有コードで定義したこのクラスがそのまま各 OS のレジストリの登録キーになる。
 *
 * @property message ダイアログに表示するメッセージ
 */
class TextInputDialogViewModel(val message: String) : DialogViewModel<String>

File: samples/maui/KsDialogs.Sample.Maui/TextInputDialogViewModel.cs
namespace KsDialogs.Sample.Maui;

/// <summary>
/// Text Input Dialog の ViewModel。
/// </summary>
/// <remarks>
/// 結果型は <see cref="string"/> で、完了操作は入力された文字列を報告する。
/// この型そのものがレジストリの登録キーになる。
/// </remarks>
/// <param name="message">ダイアログに表示するメッセージ。</param>
public sealed class TextInputDialogViewModel(string message) : IDialogViewModel<string>
{
    /// <summary>ダイアログに表示するメッセージ。</summary>
    public string Message { get; } = message;
}
